src/main/ruby/core/truffle/kernel_operations.rb

```diff
@@ -107,41 +107,41 @@ def self.define_read_only_global(name, getter)
       -> { global_variable_get(:$stderr) },
       -> v { raise TypeError, "$stderr must have a write method #{v.class} given." unless v.respond_to?(:write)
              global_variable_set(:$stderr, v) })
-   
-     def self.internal_raise(exc, msg, ctx=nil, internal)
-       skip = false
-       if undefined.equal? exc
-         exc = $!
-         if exc
-           skip = true
-         else
-           exc = RuntimeError.new('No current exception')
-         end
-       elsif exc.respond_to? :exception
-         if undefined.equal? msg
-           exc = exc.exception
-         else
-           exc = exc.exception msg
-         end
-         raise ::TypeError, 'exception class/object expected' unless exc.kind_of?(::Exception)
-       elsif exc.kind_of? String
-         exc = ::RuntimeError.exception exc
-       else
-         raise ::TypeError, 'exception class/object expected'
-       end
-
-       unless skip
-         exc.set_context ctx if ctx
-         exc.capture_backtrace!(3) unless exc.backtrace?
-         exc.instance_variable_set(:@cause, $!)
-       end
-
-       if $DEBUG
-         STDERR.puts "Exception: `#{exc.class}' #{caller.first} - #{exc.message}\n"
-       end
-
-       Truffle.invoke_primitive :vm_raise_exception, exc, internal
-     end
-   
+
+    def self.internal_raise(exc, msg, ctx=nil, internal)
+      skip = false
+      if undefined.equal? exc
+        exc = $!
+        if exc
+          skip = true
+        else
+          exc = RuntimeError.new('No current exception')
+        end
+      elsif exc.respond_to? :exception
+        if undefined.equal? msg
+          exc = exc.exception
+        else
+          exc = exc.exception msg
+        end
+        raise ::TypeError, 'exception class/object expected' unless exc.kind_of?(::Exception)
+      elsif exc.kind_of? String
+        exc = ::RuntimeError.exception exc
+      else
+        raise ::TypeError, 'exception class/object expected'
+      end
+
+      unless skip
+        exc.set_context ctx if ctx
+        exc.capture_backtrace!(3) unless exc.backtrace?
+        exc.instance_variable_set(:@cause, $!)
+      end
+
+      if $DEBUG
+        STDERR.puts "Exception: `#{exc.class}' #{caller.first} - #{exc.message}\n"
+      end
+
+      Truffle.invoke_primitive :vm_raise_exception, exc, internal
+    end
```

Comment:
This should be at the top, next to other methods.
I'm also concerned adding an extra level of call for every raise is going to have an effect on performance, as it means one more frame to capture.